File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Layers/addnewlayer.rb
module Asposediagramjava
  module AddNewLayer
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Basic_Shapes.vsd")

        # get Visio page
        page = diagram.getPages().getPage(0)

        # initialize a new Layer class object
        layer = Rjb::import('com.aspose.diagram.Layer').new
        
        # set Layer name
        layer.getName().setValue("Layer2")
        
        # set Layer Visibility
        layer.getVisible().setValue(2)

        # add Layer to the particular page sheet
        page.getPageSheet().getLayers().add(layer)

        # add a new shape
        shape_id = page.addShape(3.0, 3.0, 0.36, 0.36, "Circle")
        
        # get the Shape object
        shape = page.getShapes().getShape(shape_id)
        
        # assign this new Layer
        shape.getLayerMem().getLayerMember().setValue(layer.getIX().to_s)

        # Save diagram
        diagram.save(data_dir + "AddNewLayer.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Added a new Layer in the Visio Diagram."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithLayers/AddNewLayer.php
<?php
namespace Aspose\Diagram\WorkingwithLayers;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\Layer as Layer;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class AddNewLayer{

    public static function run($dataDir=null){
        # Create instance of Diagram
        $diagram = new Diagram($dataDir . "Basic_Shapes.vsd");

        # get Visio page
        $page = $diagram->getPages()->getPage(0);

        # initialize a new Layer class object
        $layer = new Layer();

        # set Layer name
        $layer->getName()->setValue("Layer2");

        # set Layer Visibility
        $layer->getVisible()->setValue(2);

        # add Layer to the particular page sheet
        $page->getPageSheet()->getLayers()->add($layer);

        # add a new shape
        $shape_id=$page->addShape(3.0, 3.0, 0.36, 0.36, "Circle");

        # get the Shape object
        $shape = $page->getShapes()->getShape($shape_id);

        # assign this new Layer
        $shape->getLayerMem()->getLayerMember()->setValue((string)$layer->getIX());

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir . "AddNewLayer.vdx", $saveFileFormat->VDX);

        print "Added a new Layer in the Visio Diagram.".PHP_EOL;
    }

}